### ISSUE TITLE
[in progress] Making a clean install more complete by adding some settings in the setup

### ISSUE THREAD

**Issue by gpsietzema** [modxcms/revolution#13876]
### Summary
*Please note: this is a work in progress*

A clean install of MODX is currently too clean. It's missing some very essential things, which will make the install more tailored to your needs. Therefore we need to enhance the current setup page with some more options, maybe even with an 'Make MODX more personal' button to expand these settings.

The following settings are blank or default to something very specific:

| Key | What to do with it? |
|---|---|
|date_timezone|Is currently not set, which might mess up certain timestamps.|
|cultureKey|Should be set in setup|
|fe_editor_lang|Should be set in setup|
|locale|Is currently empty and should be set in setup|
|manager_week_start|Is set to 0 (sunday) right now. Should also be able to set this to 1 (monday)|
|manager_data_format|Is set to US-version, should be setup in setup|
|manager_time_format|Is set to US-version am/pm, should be changeable to 24h format|

The following System Settings should have a different default setting:

| Key | Change |
|---|---|
| preserve_menuindex |Currently True, should be False.|
| resource_tree_node_name_fallback |Should be longtitle or menutitle|
| site_name| Is currently 'MODX Revolution', while this should really be the first thing to set for your site|
| site_status| Is set to True, while you should be able to set it at start. Websites usually start unpublished|

**Comment by alroniks** [comment]
FURLs used everywhere now and it also should be possible to turn on it during the setup (with notices about required changes in nginx/apache config files).

**Comment by JoshuaLuckers** [comment]
I would also love to see some kind of default error page. I think a lot of (new) users don't know that it's a setting. Maybe this should be a separate issue.

**Comment by Jako** [comment]
@JoshuaLuckers Good Idea: A default error_page and maybe the default unauthorized_page should be created during the setup.

**Comment by Jako** [comment]
The ht.access file should be renamed automatically by an option. A nginx config file should be prepared, if thats possible.

**Comment by meshkov** [comment]
url_scheme,  because #13269 the setting server_protocol does not do anything

**Comment by Ruslan-Aleev** [comment]
Related with https://github.com/modxcms/revolution/issues/14238 and https://github.com/modxcms/revolution/issues/14273

**Comment by Ruslan-Aleev** [comment]
It seems to me that you should not add all the fields to the installation process.
It would be more correct to change the default settings, for example, for days and time, since most countries use others.

**Comment by JoshuaLuckers** [comment]
The `manager_time_format` is now using the 24-hour clock format after merging PR #14325

**Comment by Jako** [comment]
`fe_editor_lang` is deprecated according to https://docs.modx.com/revolution/2.x/administering-your-site/settings/system-settings/fe_editor_lang

**Comment by muzzwood** [comment]
This is taking it even further but I would love MODX 3 to provide different installation and configuration options so it doesn't have to just be a blank canvas. (This is daunting for newbies.)

Options such as:
- Multilingual Site: Could be done similarly to an extra I released recently called MultiLangSetup.
- Configure a Blog
- News Site
- Event Listings

Of course one size doesn't fit all but they could using the best practices from experienced MODX developers. Users can still build things their own way and not use these configurations.

Note that this is not adding things into the core besides the configuration scripts. Each configuration should rely on existing extras (e.g. Babel and LangRouter for configuring a multilingual site, or Collections, Tickets etc for a blog.)

If someone doesn't want these configurations they just don't select the option.

Thoughts?

**Comment by Ibochkarev** [comment]
@digitalpenguin I had the same thoughts on this account. Need a discussion with the community on this issue.

**Comment by ghost** [comment]
Composer and dependencies would make it much easier, there wouldn't need to be several different distributions, nor would there need to be unused packages.

**Comment by Ruslan-Aleev** [comment]
@digitalpenguin There are a few comments on this idea:
1) If you give the user 4 categories to choose from, the user will in the future think only of these categories and not be aware of all the flexibility of MODX. Although maybe for beginners it will be more useful.
2) The idea with the choice of packages is interesting, there was a similar discussion earlier. But what if, for example, a vulnerability appeared in Babel? How the package configuration will work then? The configuration somehow needs to be built dynamically proceeding from a rating of packages from https://modx.com/extras/ with the ability to remove or add the required package.

**Comment by ghost** [comment]
I would presume that the dependencies would be downloaded on demand, not built in to the whole MODX package. So any vulnerabilities would get fixed, and the latest version would be fetched.

**Comment by Ibochkarev** [comment]
@digitalpenguin I propose to make a separate issue.

**Comment by muzzwood** [comment]
@Ruslan-Aleev Yes I agree, the 4 options I gave were just arbitrary examples. Exactly what options are included would need to be decided upon.
In my mind the dependencies would be downloaded on demand, like what @sottwell mentioned so it would be important to select Extras that are actively maintained.
@Ibochkarev Sure, I'll create a separate issue for it now. Thanks!

**Comment by Ibochkarev** [comment]
I propose to summarize what will be included in the installer.

**Comment by Ruslan-Aleev** [comment]
If I’m not mistaken, all the date and time settings will be taken automatically based on the selected language, @Alroniks mentioned this.

I would add only 2 additional settings fields:

`site_status and site_name`

We should not add a lot of new fields, because tired to fill them in when you install MODX :)

For the rest of the settings, for example, _error_page or publish_default_, I would not include them in the installation, but at the last stage of the installation should be displayed a message that

`the site should be finally configured through the section "System Settings -> Site" or through other sections of the "System Settings"` (with link to the System Settings section) 

@JoshuaLuckers, @Mark-H, @Jako have any ideas?

**Comment by JoshuaLuckers** [comment]
> I would add only 2 additional settings fields:
> 
> `site_status and site_name`
> We should not add a lot of new fields, because tired to fill them in when you install MODX :)

This would get my approval. 
 
> For the rest of the settings, for example, _error_page or publish_default_, I would not include them in the installation, but at the last stage of the installation should be displayed a message that
> 
> `the site should be finally configured through the section "System Settings -> Site" or through other sections of the "System Settings"`

[Jason mentioned this in another issue:](https://github.com/modxcms/revolution/issues/14541#issuecomment-480859719)
> Moving forward, i.e. for 3.x+, I think we should come up with a simple app to present the welcome information from modx.org with an i18n-capable presentation that could be maintained by contributors, similar to the documentation efforts.

How to "complete configuring your website" should be part of this "app".

**Comment by Ruslan-Aleev** [comment]
I would add information "complete configuring your website" both in the "app" and in the installer :)
